plugins/cost-insights/src/components/CostOverviewCard/CostOverviewByProduct.tsx

```diff
@@ -0,0 +1,211 @@
+  XAxis,
+  YAxis,
+  Tooltip as RechartsTooltip,
+  Area,
+  ResponsiveContainer,
+  CartesianGrid,
+} from 'recharts';
+import { Cost, DEFAULT_DATE_FORMAT, CostInsightsTheme } from '../../types';
+import {
+  BarChartTooltip as Tooltip,
+  BarChartTooltipItem as TooltipItem,
+} from '../BarChart';
+import {
+  overviewGraphTickFormatter,
+  formatGraphValue,
+  isInvalid,
+} from '../../utils/graphs';
+import {
+  useCostOverviewStyles as useStyles,
+  DataVizColors,
+} from '../../utils/styles';
+import { useFilters, useLastCompleteBillingDate } from '../../hooks';
+import { mapFiltersToProps } from './selector';
+import { getPreviousPeriodTotalCost } from '../../utils/change';
+import { LegendItem } from '../LegendItem';
+import { currencyFormatter } from '../../utils/formatters';
+import { aggregationSum } from '../../utils/sum';
+
+dayjs.extend(utc);
+
+export type CostOverviewByProductProps = {
+  dailyCostData: Cost;
+};
+
+const LOW_COST_THRESHOLD = 0.01;
+
+export const CostOverviewByProduct = ({
+  dailyCostData,
+}: CostOverviewByProductProps) => {
+  const theme = useTheme<CostInsightsTheme>();
+  const styles = useStyles(theme);
+  const lastCompleteBillingDate = useLastCompleteBillingDate();
+  const { duration } = useFilters(mapFiltersToProps);
+  const groupedCosts = dailyCostData.groupedCosts;
+
+  if (!groupedCosts) {
+    return null;
+  }
+
+  const flattenedAggregation = groupedCosts
+    .map(cost => cost.aggregation)
+    .flat();
+  const totalCost = flattenedAggregation.reduce(
+    (acc, agg) => acc + agg.amount,
+    0,
+  );
+
+  const productsByDate = groupedCosts.reduce((prodByDate, group) => {
+    const product = group.id;
+    group.aggregation.forEach(curAggregation => {
+      const productCostsForDate = prodByDate[curAggregation.date] || {};
+      // Group products with less than 10% of the total cost into "Other" category
+      if (aggregationSum(group.aggregation) < totalCost * LOW_COST_THRESHOLD) {
+        prodByDate[curAggregation.date] = {
+          ...productCostsForDate,
+          Other:
+            (prodByDate[curAggregation.date].Other || 0) +
+            curAggregation.amount,
+        };
+      } else {
+        prodByDate[curAggregation.date] = {
+          ...productCostsForDate,
+          [product]: curAggregation.amount,
+        };
+      }
+    });
+
+    return prodByDate;
+  }, {} as Record<string, Record<string, number>>);
+
+  const chartData: any[] = Object.keys(productsByDate).map(date => {
+    return {
+      ...productsByDate[date],
+      date: Date.parse(date),
+    };
+  });
+
+  const tooltipRenderer: ContentRenderer<TooltipProps> = ({
+    label,
+    payload = [],
+  }) => {
+    if (isInvalid({ label, payload })) return null;
+
+    const title = dayjs(label).utc().format(DEFAULT_DATE_FORMAT);
+    const items = payload.map(p => ({
+      label: p.dataKey as string,
+      value: formatGraphValue(p.value as number),
+      fill: p.fill!,
+    }));
+
+    return (
```

Comment:
If you haven't already, do a UI sanity check on the tooltip for products with very large costs